lib/mongo/crypt/binary.rb

```diff
@@ -50,6 +63,60 @@ def initialize(data=nil)
         end
       end
 
+      # Initialize a Binary object from an existing pointer to a mongocrypt_binary_t
+      # object.
+      #
+      # @param [ FFI::Pointer ] pointer A pointer to an existing
+      #   mongocrypt_binary_t object
+      #
+      # @return [ Mongo::Crypt::Binary ] A new binary object
+      def self.from_pointer(pointer)
+        self.new(pointer: pointer)
+      end
+
+      # Initialize a Binary object with a string. The Binary object will store a
+      # copy of the specified string and destroy the allocated memory when
+      # it goes out of scope.
+      #
+      # @param [ String ] data A string to be wrapped by the Binary object
+      #
+      # @return [ Mongo::Crypt::Binary ] A new binary object
+      def self.from_data(data)
+        self.new(data: data)
+      end
+
+      # Overwrite the existing data wrapped by this Binary object
+      #
+      # @note The data passed in must not take up more memory than the
+      # original memory allocated to the underlying mongocrypt_binary_t
+      # object. Do NOT use this method unless required to do so by libmongocrypt.
+      #
+      # @param [ String ] data The new string data to be wrapped by this binary object
+      #
+      # @return [ true | false ] Whether the operation was successful
+      def write(data)
+        # Cannot write a string that's longer than the space currently allocated
+        # by the mongocrypt_binary_t object
+        return false if Binding.mongocrypt_binary_len(@bin) < data.length
```

Comment:
How come this does not raise an exception?

Reply:
Good question. That would actually make more sense. I'll change it to do that right now.